### PR TITLE
BTF: use btf_dump__new_deprecated when necessary

### PR DESCRIPTION
**Pull request by viktormalik**
libbpf 0.6.0 has automatic detection of which `btf_dump__new` version to use, however it uses some C-specific features and doesn't work for C++. More details can be found in #2068.

Therefore, to be able support all older versions of libbpf, we need to detect presence of `btf_dump__new_deprecated` and use that one instead of `btf_dump__new`, if it's available.

This should make bpftrace compilable on systems with libbpf 0.6.0 (e.g. Fedora 35). It's a rather specific use-case, but 0.6.0 is not that old (F35 is still supported), so I think we should take this.

##### Checklist

- [ ] Language changes are updated in `man/adoc/bpftrace.adoc` and if needed in `docs/reference_guide.md`
- [ ] User-visible and non-trivial changes updated in `CHANGELOG.md`
- [ ] The new behaviour is covered by tests

### CODE REVIEW

**Review by martinetd — CHANGES_REQUESTED** [review]
~~I'd agree we should try to fix this if we could, but I think this is hopeless -- see full rationale in https://github.com/iovisor/bpftrace/issues/2068 . Sorry for anyone subscribed to both the issue and PRs, just want to make sure this is seen here if someone only checks here.

(by the way `btf_dump__new_deprecated` and `btf_dump__new_v0_6_0` got added in the same commit)~~

(EDIT: also striking that out for clarity to anyone reading this now)

File: src/btf.cpp [hunk 1]
```diff
@@ -204,9 +204,12 @@ static struct btf_dump *dump_new(const struct btf *btf,
   struct btf_dump_opts opts = {
```
[the remark this answers is not in the document]
Comment (viktormalik) [reply src/btf.cpp:204]: So, the verdict is that the current fix is correct or not? :-)

[the remark this answers is not in the document]
Comment (martinetd) [reply src/btf.cpp:204]: I think it's ok: This should work pre-0.6 (no new symbol, both defines false), 0.6 (only deprecated symbol "found"), 0.7/0.8 (both found, new one used) and master (only new version should be found and used despite not being named _v0_6_0 anymore).

I didn't test though.

[the remark this answers is not in the document]
Comment (viktormalik) [reply src/btf.cpp:204]: We have CI tests for libbpf 0.5 and 0.8. We could theoretically add 0.6 to test this particular feature but it seems to me a too specific case to run the entire test suite.

**Review by fbs — APPROVED** [review]

**Review by danobi — APPROVED** [review]

**Pull request merged**